packages/core/src/checkout-buttons/strategies/googlepay/googlepay-button-strategy.ts

```diff
@@ -116,12 +139,44 @@ export default class GooglePayButtonStrategy implements CheckoutButtonStrategy {
         throw new InvalidArgumentError();
     }
 
+    private async _createBuyNowCart({ buyNowInitializeOptions }: BuyNowInitializeOptions) {
+        if (typeof buyNowInitializeOptions?.getBuyNowCartRequestBody === 'function' && this._cartRequestSender) {
+            const cartRequestBody = buyNowInitializeOptions.getBuyNowCartRequestBody();
+
+            if (!cartRequestBody) {
+                throw new MissingDataError(MissingDataErrorType.MissingCart);
+            }
+
+            try {
+                const { body: cart } = await this._cartRequestSender.createBuyNowCart(cartRequestBody);
+
+                return cart;
+            } catch (error) {
+                throw new BuyNowCartCreationError();
+            }
+        }
+    }
+
     @bind
-    private async _handleWalletButtonClick(event: Event): Promise<void> {
+    private async _handleWalletButtonClick(event: Event, { buyNowInitializeOptions }: BuyNowInitializeOptions, currencyCode?: string): Promise<void> {
         event.preventDefault();
-        const cart = this._store.getState().cart.getCartOrThrow();
+
+        this._buyNowCart = await this._createBuyNowCart({ buyNowInitializeOptions });
+
+        const cart = this._buyNowCart || this._store.getState().cart.getCartOrThrow();
```

Comment:
all of this _cartRequestSender logic work only with braintree. Is it possible to move this logic to the googlepay braintree processor?
I believe we can somehow update cart from braintree processor and do not modify googlepay-button-strategy.ts.
@animesh1987 what do you think about that?

Reply:
resolved